lib/jsdom/living/post-message.js

```diff
@@ -22,11 +23,28 @@ module.exports = function (message, targetOrigin) {
     return;
   }
 
+  // Get call stack from Error object, so we can know which file (uri) invoked this method
+  const stack = new Error().stack.split("\n");
+  const prevStack = stack[2];
+
+  let sourceOrigin = this.location.origin;
+
+  // grab the filename part from stack trace and the hostname should be the origin
```

Comment:
This is not really true in several different ways.

1. If an HTML file on domain1.com includes a script from domain2.com (using `<script src>`), then the origin-in-effect (the [incumbent](https://html.spec.whatwg.org/multipage/webappapis.html#concept-incumbent-everything) origin, per [window post message steps](https://html.spec.whatwg.org/multipage/web-messaging.html#window-post-message-steps) step 8.2) should in fact be domain1.com.
2. The current treatment doesn't take into account port numbers, which are generally part of an [origin](https://html.spec.whatwg.org/multipage/origin.html#concept-origin-tuple).

I'm wondering if the best we could do is to always use the target origin as the source origin.

----

Also, please name your variables as similarly as possible to [the spec](https://html.spec.whatwg.org/multipage/web-messaging.html#window-post-message-steps).

Reply:
Thanks @TimothyGu for and advices, I will change the variable naming and try to figure out if there's any solution on these situations.